main/src/ca/uwaterloo/flix/language/phase/njvm/classes/TupleClass.scala

```diff
@@ -0,0 +1,242 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language.phase.njvm.classes
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.FinalAst.Root
+import ca.uwaterloo.flix.language.phase.jvm.{JvmClass, JvmName}
+import ca.uwaterloo.flix.language.phase.njvm.Mnemonics.MnemonicsTypes.{MArray, _}
+import ca.uwaterloo.flix.language.phase.njvm.Mnemonics._
+import ca.uwaterloo.flix.language.phase.njvm.Mnemonics.Instructions._
+import ca.uwaterloo.flix.language.phase.njvm.NJvmType
+import ca.uwaterloo.flix.language.phase.njvm.NJvmType.{PrimBool, PrimByte, PrimChar, PrimDouble, PrimFloat, PrimInt, PrimLong, PrimShort, Reference}
+import ca.uwaterloo.flix.util.InternalCompilerException
+
+import scala.reflect.runtime.universe._
+
+class TupleClass(map : Map[JvmName, MnemonicsClass], elms : List[NJvmType])(implicit root: Root, flix : Flix) extends MnemonicsClass {
+  //Setup
+  private val ct: Reference = getTupleClassType(elms)
+  private val cg: ClassGenerator = new ClassGenerator(ct, List(getTupleInterfaceType(elms)))
+
+  //Fields
+  for ((arg, ind) <- elms.zipWithIndex) {
+      arg match {
+        case PrimBool => cg.mkPrimField[MBool]("field" + ind)
+        case PrimChar => cg.mkPrimField[MChar]("field" + ind)
+        case PrimByte => cg.mkPrimField[MByte]("field" + ind)
+        case PrimShort => cg.mkPrimField[MShort]("field" + ind)
+        case PrimInt => cg.mkPrimField[MInt]("field" + ind)
+        case PrimLong => cg.mkPrimField[MLong]("field" + ind)
+        case PrimFloat => cg.mkPrimField[MFloat]("field" + ind)
+        case PrimDouble => cg.mkPrimField[MDouble]("field" + ind)
+        case Reference(_) => cg.mkField[Ref[MObject]]("field" + ind)
+        case _ => throw InternalCompilerException(s"Unexpected type $arg")
+      }
+  }
+
+  private def getField[T1 <: MnemonicsTypes : TypeTag](ind : Int) : Field[T1] =
+    new Field[T1]("field" + ind)
+
+  private def getPrimField[T1 <: MnemonicsPrimTypes : TypeTag](ind : Int) : PrimField[T1] =
+    new PrimField[T1]("field" + ind)
+
+  val defaultConstrutor : UncheckedVoidMethod = {
+    val setFields =
+      (sig : UncheckedFunSig)  =>{
+        var ins = NO_OP[StackNil]
+        var offset: Int = 1
+        for ((arg,ind) <- elms.zipWithIndex) {
+          ins = ins |>>
+          sig.getArg[Ref[TupleClass]](0).LOAD |>>
+          (arg match {
+            case PrimBool => sig.getArg[MBool](offset).LOAD[StackNil ** Ref[TupleClass]] |>>
```

Comment:
Add a linebreak after => so it is easier to see that there are two instructions (LOAD, followed by PUT_FIELD). Do this for all the cases.